src/quickgui/qgsquickpositionkit.cpp

```diff
@@ -103,28 +117,42 @@ void QgsQuickPositionKit::use_gps_location()
 
 void QgsQuickPositionKit::replacePositionSource( QGeoPositionInfoSource *source )
 {
-  if ( mSource == source )
+  if ( mSource.get() == source )
     return;
 
   if ( mSource )
   {
-    disconnect( mSource, nullptr, this, nullptr );
-    delete mSource;
-    mSource = nullptr;
+    disconnect( mSource.get(), nullptr, this, nullptr );
+    mSource.reset();
   }
 
-  mSource = source;
+  mSource.reset( source );
 
   if ( mSource )
   {
-    connect( mSource, &QGeoPositionInfoSource::positionUpdated, this, &QgsQuickPositionKit::positionUpdated );
-    connect( mSource, &QGeoPositionInfoSource::updateTimeout, this,  &QgsQuickPositionKit::onUpdateTimeout );
+    connect( mSource.get(), &QGeoPositionInfoSource::positionUpdated, this, &QgsQuickPositionKit::positionUpdated );
+    connect( mSource.get(), &QGeoPositionInfoSource::updateTimeout, this,  &QgsQuickPositionKit::onUpdateTimeout );
     mSource->startUpdates();
 
     QgsDebugMsg( QStringLiteral( "Position source changed: %1" ).arg( mSource->sourceName() ) );
   }
 }
 
+void QgsQuickPositionKit::setCoordinateTransformMapSettings()
+{
+  coordinateTransformer()->setMapSettings( mapSettings() );
+}
+
+QgsQuickMapSettings *QgsQuickPositionKit::mapSettings() const
+{
+  return mMapSettings;
+}
+
+QgsQuickCoordinateTransformer *QgsQuickPositionKit::coordinateTransformer() const
```

Comment:
coordinate transformer should be private, not exposed through API at all... remove this function.